File: app/src/main/java/com/project/smartmunimji/SellerProductsActivity.java
package com.project.smartmunimji;

import android.app.AlertDialog;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import com.project.smartmunimji.databinding.ActivitySellerProductsBinding;
import com.project.smartmunimji.model.Product;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class SellerProductsActivity extends AppCompatActivity {

    private ActivitySellerProductsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySellerProductsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        String sellerId = getIntent().getStringExtra("SELLER_ID");
        String sellerName = getIntent().getStringExtra("SELLER_NAME");
        binding.sellerProductsLabel.setText(getString(R.string.products_from_seller, sellerName));

        // Mock products based on seller
        List<Product> products = new ArrayList<>();
        if ("1".equals(sellerId)) { // TechMart
            products.add(new Product("Laptop", "2023-01-15", "2024-01-15", true));
            products.add(new Product("Mouse", "2023-06-10", "2023-12-10", false));
        } else if ("2".equals(sellerId)) { // HomeGoods
            products.add(new Product("Blender", "2023-03-20", "2024-03-20", true));
            products.add(new Product("Toaster", "2022-11-05", "2023-11-05", false));
        } else { // Book Nook
            products.add(new Product("Book", "2023-04-01", "2024-04-01", true));
        }

        ProductAdapter adapter = new ProductAdapter(products, product -> {
            Toast.makeText(this, "Claim Warranty clicked for " + product.getName(), Toast.LENGTH_SHORT).show();
            AlertDialog.Builder builder = new AlertDialog.Builder(this);
            builder.setTitle("Claim Warranty");
            android.widget.EditText input = new android.widget.EditText(this);
            input.setHint(R.string.issue_description);
            builder.setView(input);
            builder.setPositiveButton(R.string.submit, (dialog, which) -> {
                Toast.makeText(this, "Claim submitted. Seller notified!", Toast.LENGTH_LONG).show();
            });
            builder.setNegativeButton("Cancel", null);
            builder.show();
        });
        binding.productsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        binding.productsRecyclerView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/project/smartmunimji/SellerProductsActivity.kt
// app/src/main/java/com/project/smartmunimji/SellerProductsActivity.kt

package com.project.smartmunimji

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.project.smartmunimji.adapters.ProductAdapter
import com.project.smartmunimji.databinding.ActivitySellerProductsBinding
import com.project.smartmunimji.model.response.ProductListResponse
import com.project.smartmunimji.network.RetrofitClient
import com.project.smartmunimji.repository.AppRepository
import com.project.smartmunimji.utils.TokenManager
import com.project.smartmunimji.viewmodel.Status
import com.project.smartmunimji.viewmodel.ViewAllProductsViewModel
import com.project.smartmunimji.viewmodel.ViewAllProductsViewModelFactory
import java.util.ArrayList

class SellerProductsActivity : AppCompatActivity(), ProductAdapter.OnProductClickListener {

    private lateinit var binding: ActivitySellerProductsBinding
    private lateinit var productsViewModel: ViewAllProductsViewModel
    private lateinit var productAdapter: ProductAdapter
    private lateinit var tokenManager: TokenManager

    private var selectedSellerName: String? = null // Seller name passed via Intent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySellerProductsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(findViewById(R.id.toolbar))
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        selectedSellerName = intent.getStringExtra("SELLER_NAME")
        binding.sellerNameText.text = selectedSellerName?.let {
            getString(R.string.products_from_seller, it)
        } ?: getString(R.string.products_from_seller, "Unknown Seller")


        tokenManager = TokenManager(this)

        // Initialize ViewModel (re-using ViewAllProductsViewModel)
        val apiService = RetrofitClient.getApiService(tokenManager)
        val repository = AppRepository(apiService)
        val viewModelFactory = ViewAllProductsViewModelFactory(repository)
        productsViewModel = ViewModelProvider(this, viewModelFactory).get(ViewAllProductsViewModel::class.java)

        // Setup RecyclerView
        productAdapter = ProductAdapter(emptyList(), this)
        binding.sellerProductsRecyclerView.apply {
            layoutManager = LinearLayoutManager(this@SellerProductsActivity)
            adapter = productAdapter
        }

        // --- Observe LiveData from ViewModel ---

        productsViewModel.products.observe(this) { allProducts ->
                val filteredProducts = if (selectedSellerName != null) {
            allProducts.filter { it.getShopName() == selectedSellerName }
        } else {
            allProducts
        }
            productAdapter.updateProducts(filteredProducts)
            binding.emptyStateText.visibility = if (filteredProducts.isEmpty()) View.VISIBLE else View.GONE
        }

        productsViewModel.fetchStatus.observe(this) { status ->
                when (status) {
            is Status.Loading -> {
                binding.progressBar.visibility = View.VISIBLE
                binding.sellerProductsRecyclerView.visibility = View.GONE
                binding.emptyStateText.visibility = View.GONE
            }
            is Status.Success -> {
                binding.progressBar.visibility = View.GONE
                binding.sellerProductsRecyclerView.visibility = View.VISIBLE
                // Empty state handled by products observer
            }
            is Status.Error -> {
                binding.progressBar.visibility = View.GONE
                binding.sellerProductsRecyclerView.visibility = View.GONE
                binding.emptyStateText.visibility = View.VISIBLE
                binding.emptyStateText.text = status.message
                Toast.makeText(this, status.message, Toast.LENGTH_LONG).show()
            }
        }
        }

        // --- Initial Data Fetch ---
        productsViewModel.fetchCustomerProducts()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    // --- ProductAdapter.OnProductClickListener implementations ---
    override fun onProductClick(product: ProductListResponse) {
        Toast.makeText(this, "Clicked on: ${product.getProductName()}", Toast.LENGTH_SHORT).show()
        // Implement navigation to a product detail page if desired
    }

    override fun onClaimWarrantyClick(product: ProductListResponse) {
        if (product.isWarrantyEligible()) {
            val intent = Intent(this, ClaimWarrantyActivity::class.java).apply {
                putExtra("REGISTERED_PRODUCT_ID", product.getRegisteredProductId())
                putExtra("PRODUCT_NAME", product.getProductName())
            }
            startActivity(intent)
        } else {
            Toast.makeText(this, "Warranty for ${product.getProductName()} is expired.", Toast.LENGTH_SHORT).show()
        }
    }
}